airbyte-cdk/java/airbyte-cdk/build.gradle

```diff
@@ -0,0 +1,76 @@
+plugins {
+    id 'java-library'
+    id 'maven-publish'
+}
+
+group 'io.airbyte'
+
+// Version is dynamically loaded from version.properties file.
+def props = new Properties()
+file("src/main/resources/version.properties").withInputStream { props.load(it) }
+version = props.getProperty('version')
+description = "Airbyte Connector Development Kit (CDK) for Java."
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.7.2'
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = 'io.airbyte'
+            artifactId = 'airbyte-cdk'
+            from components.java
+        }
+    }
+    repositories {
+        maven {
+            name 'cloudrepo'
+            url 'https://airbyte.mycloudrepo.io/repositories/airbyte-public-jars'
+            credentials {
+                username System.getenv('CLOUDREPO_USER')
+                password System.getenv('CLOUDREPO_PASSWORD')
+            }
+        }
+    }
+}
+
+// Adds publishToMavenLocal as final command in the list of 'build' tasks.
+build.dependsOn(publishToMavenLocal)
```

Comment:
one implication here, as discussed in the meeting, is that this now overloads what build does which may be non-intuitive to people who use gradle a lot.

We should either separate this into a separate task (maybe like buildAndLocalPublish) or be explicit with logging, errors, etc that is what we are doing

Reply:
The other kind of side effect here is that, afaik, the way build works, if you do a top level build, only this one sub-build will have this separate behavior, so like, if I do ./gradlew build at the top level I'll have a bunch of stuff that does the normal build thing (dump resources into the build folder and run tests etc) and then this one subtask that does an extra thing, publish to mavenlocal(). Which again I think is relatively fine, as long as we make that behavior obvious.

One thing to make sure we test, actually, does this build as we expect if we bump the version and then run ./gradlew SUB_BUILD=connectors_base? My understanding is that bumping cdk version and a connector version and then running a top level build will fail, and that is expected behavior (because of the dual gradle task dependency thing we talked about previously, gradle needs the dependency to be built already before it can resolve it)